File: 0066_plus_one/plus_one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int n = digits.length;
        int i = n - 1;
        while (i >= 0 && digits[i] == 9) {
            digits[i] = 0;
            i--;
        }
        if (i < 0) {
            int res[] = new int[n + 1];
            res[0] = 1;
            return res;
        }
        digits[i]++;
        return digits;
    }
}

File: 0066_plus_one/plus_one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        i = len(digits) - 1
        while digits[i] == 9:
            digits[i] = 0
            i -= 1
        if i < 0:
            return [1] + digits
        digits[i] += 1
        return digits
    
    
    def plusOneComprehensions(self, digits: List[int]) -> List[int]:
        num = int("".join([str(s) for s in digits]))
        return [s for s in str(num + 1)]